plugins/onepad/state_management.cpp

```diff
@@ -248,28 +248,27 @@ u8 pad_poll(u8 value)
 
                 if (pad->mode != MODE_DIGITAL) { // ANALOG || DS2 native
                     query.numBytes = 9;
-
-                    query.response[5] = g_key_status.get(query.port, PAD_R_RIGHT);
-                    query.response[6] = g_key_status.get(query.port, PAD_R_UP);
-                    query.response[7] = g_key_status.get(query.port, PAD_L_RIGHT);
-                    query.response[8] = g_key_status.get(query.port, PAD_L_UP);
+                    buttons, PAD_R_RIGHT;
```

Comment:
I guess you should redo the code here. I would advice to test your change locally before the push.

So current situation is rather bad. Technically we should update all `KeyStatus` code to use an array of array (port, slot). As a temporary solution we could do
```
// Build a linear pad number (range from 0 to 3) from the slot and the port. Technically there are 4 slots but KeyStatus code support only 4 gamepads (2 ports * 2 slots).
int pad_number = query.port + 2 * query.slot;
```
Then you could use this kind of code
```
query.response[5] = g_key_status.get(pad_number, PAD_R_RIGHT);
...
```

Reply:
Done, still no changes in functionality.  players 1 and 2 work like normal, if multitap is set, that controller controls that slot and the slots after it.  leads me to believe that setting multitap (either) is the source of the biggest blocker.  I'll keeping reading through and learning, Thanks for the issues you've found, you've been a big help.

Reply:
No you misunderstood me.
```
int pad_number = query.port + 2 * query.slot;
```
pad number must be updated after query update. Here the value is set once. Basically you need to compute the pad_number after
```
Pad *pad = &pads[query.port][query.slot];
```

By the way why did you remove test_bit test ?

Reply:
ahhh, still learning, C++ is quite different from java in a sickening number of ways.

Your example of 
"query.response[5] = g_key_status.get(pad_number, PAD_R_RIGHT);"
didn't have the !testbit part in it, so I mistakenly assumed that it should have been coded in that manner instead.  

I'll fix it up sir :)

The part that makes me wonder is nothing has functionally changed after any iteration of changes.  Could it be that this is legacy code from Onepad original that is superceded by the SDL2 code?